core/src/main/java/org/apache/brooklyn/core/objs/BasicConfigurableObject.java

```diff
@@ -53,7 +53,7 @@ public BasicConfigurableObject() {
     }
 
     @Override
-    public void injectManagementContext(ManagementContext managementContext) {
+    public void setManagementContext(ManagementContext managementContext) {
```

Comment:
Breaks binary + source compatibility, but I guess it's fine as we've already have another precedent for this release. Worth adding to the release notes.

Out of curiosity, is this needed for OSGi injection, which library will be able to inject the context here?